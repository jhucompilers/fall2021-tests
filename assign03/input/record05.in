PROGRAM person;
  TYPE Person = RECORD
    name: ARRAY 10 OF CHAR;
    age: INTEGER;
  END;
  TYPE Parent = RECORD
    details: Person;
    children: ARRAY 2 OF Person;
  END;
  VAR parents : ARRAY 4 OF Parent;
  VAR i, j, k : INTEGER;


BEGIN
  i := 0;
  WHILE i < 4 DO
    k := 0;
    WHILE k < 10 DO
      READ parents[i].details.name[k];
    END;
    READ parents[i].details.age;
    j := 0;
    WHILE j < 2 DO
      k := 0;
      WHILE k < 10 DO
        READ parents[i].children[j].name[k];
      END;
      READ parents[i].children[j].name;
    END;

  END;
END.
