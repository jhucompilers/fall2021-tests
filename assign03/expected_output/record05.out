1,VAR,name,ARRAY 10 OF CHAR
1,VAR,age,INTEGER
0,TYPE,Person,RECORD (ARRAY 10 OF CHAR x INTEGER)
1,VAR,details,RECORD (ARRAY 10 OF CHAR x INTEGER)
1,VAR,children,ARRAY 2 OF RECORD (ARRAY 10 OF CHAR x INTEGER)
0,TYPE,Parent,RECORD (RECORD (ARRAY 10 OF CHAR x INTEGER) x ARRAY 2 OF RECORD (ARRAY 10 OF CHAR x INTEGER))
0,VAR,parents,ARRAY 4 OF RECORD (RECORD (ARRAY 10 OF CHAR x INTEGER) x ARRAY 2 OF RECORD (ARRAY 10 OF CHAR x INTEGER))
0,VAR,i,INTEGER
0,VAR,j,INTEGER
0,VAR,k,INTEGER
